php_phongo.c

```diff
@@ -1561,11 +1553,6 @@ static void php_phongo_cursor_iterator_dtor(zend_object_iterator *iter TSRMLS_DC
 {
 	php_phongo_cursor_iterator *cursor_it = (php_phongo_cursor_iterator *)iter;
 
-	if (cursor_it->cursor->firstBatch) {
-		bson_clear(&cursor_it->cursor->firstBatch);
-		cursor_it->cursor->firstBatch = NULL;
-	}
-
 	php_phongo_cursor_free_current(cursor_it->cursor);
```

Comment:
cursor_it->cursor is the `intern` of the cursor_it->intern.data zval (which is destroyed one line below), isn't it?
So yes, it must be freed first.